leetcode/medium/498._Diagonal_Traverse.py

```diff
@@ -0,0 +1,67 @@
+# Link to problem: https://leetcode.com/problems/diagonal-traverse/description/
+# Loom explaining solution you wrote (5m): TODO
+# Solution:
+# Description
+# Key ideas
+# Reason through the boundary conditions by drawing a bigger square with arrows crossing through them
+# There’s nothing difficult but one needs to be extra careful
+# Difficulty: Medium
+# Code
+
+class Solution:
+    def findDiagonalOrder(self, mat: List[List[int]]) -> List[int]:
+        def next_up_cord(cord):
+            x, y = cord
+            return [x + 1, y - 1]
+
+        def next_down_cord(cord):
+            x, y = cord
+            return [x - 1, y + 1]
+
+        def out_of_bound(cord):
+            x, y = cord
+            if x < 0 or y < 0:
+                return True
+            if y >= len(mat):
+                return True
+            if x >= len(mat[0]):
+                return True
+            return False
+
+        def at(cord):
+            x, y = tuple(cord)
+            return mat[y][x]
+
+        up = True
```

Comment:
What does this variable represent? It's unclear and should be made more clear by the name. I believe this refers the direction that you're traveling

Reply:
renamed to goUp